src/background/lotus-client/Lotus.js

```diff
@@ -47,131 +63,381 @@ class Lotus {
+        jsonrpc: "2.0",
+        method: "Filecoin.StateReadState",
+        id: 1,
+        params: [pch, null]
+      }, { headers });
+    } catch (error) {
+      ports.postLog(`ERROR: Lotus.stateReadState(): axios error: ${error.message}`);
+      return undefined
     }
-
-    return result;
+		//ports.postLog(`response.data = ${inspect(response.data)}`);
+    ports.postLog(`--------------------------------------------------------------------------------------------------`);
+    ports.postLog(`DEBUG: Lotus.stateReadState => ${inspect(response.data)} ; ${inspect(response.data.result)}`);
+    ports.postLog(`--------------------------------------------------------------------------------------------------`);
+    return response.data;
   }
 
-  async getNextNonce() {
-    const nonce = await this.post('Filecoin.MpoolGetNonce', [this.wallet]);
-    return nonce + 1;
+  /**
+   * Creates a payment voucher for the given amount valid on the given payment channel
+   * @param  {string} pch The address of the payment channel
+   * @param  {number} amountAttoFil Voucher amount in attofil
+   * @param  {number} nonce Integer that should increment for each new sv on a given payment channel
+   * @return {string} Returns the voucher as a string, or undefined if an error occurred
+   */
+  async createSignedVoucher(pch, amountAttoFil, nonce) {
+    try {
+      ports.postLog(`DEBUG: Lotus.createSignedVoucher: args: pch='${pch}', amountAttoFil='${amountAttoFil}', nonce='${nonce}'`);
+      let voucher = signer.createVoucher(pch, BigInt(0), BigInt(0), `${amountAttoFil}`, BigInt(0), BigInt(nonce), BigInt(0));
+      let signedVoucher = signer.signVoucher(voucher, this.privateKeyBase64);
+      ports.postLog(`DEBUG: Lotus.createSignedVoucher: returning signedVoucher = '${inspect(signedVoucher)}'`);
+      return signedVoucher;
+    } catch (error) {
+      ports.postLog(`ERROR: Lotus.createSignedVoucher: error: ${error.message}`);
+      return undefined
+    }
   }
 
-  async waitForMessage(messageLink) {
-    // FIXME: use websockets instead of pooling
-    let keepPooling = true;
-
-    setTimeout(() => {
-      keepPooling = false;
-    }, 10 * 1000);
+  /**
+   * Creates a new payment channel using the local Client wallet as the "From", and
+   * the specified "To" address.
+   * @param  {string} to The "To" address on the payment channel
+   * @return {number} Returns the new PCH's robust address, or undefined if an error occurred
+   */
+  async createPaymentChannel(to, amountAttoFil) {
+    const fromAddr = this.wallet
+    const fromKey = this.privateKeyBase64
+    const toAddr = to
+
+    ports.postLog(`DEBUG: Lotus.createPaymentChannel: [from:${fromAddr}, fromKey:*******************, to:${toAddr}, amount:${amountAttoFil}]`);
+
+    let nonce = await this.getNonce(fromAddr);
+    ports.postLog(`DEBUG: Lotus.createPaymentChannel: nonce=${nonce}`);
+
+    //
+    // Generate the PCH create message
+    //
+    var signedCreateMessage;
```

Comment:
Don't need this.

Reply:
Not a JS expert, so I honestly don't know, but if I declare it inside the `try {...}` block, won't it go out of scope at the end of the `try{}catch{}`?  I need to use that value later, after the try catch.

For now marked this line as a `// TODO:  does this need to be declared out here?`